src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -2808,7 +2814,7 @@ mongoc_cluster_stream_for_reads (mongoc_cluster_t *cluster,
    }
 
    return _mongoc_cluster_stream_for_optype (
-      cluster, MONGOC_SS_READ, prefs_override, cs, reply, error);
+      cluster, has_write_stage, prefs_override, cs, reply, error);
```

Comment:
`has_write_stage` is being converted to `mongoc_ss_optype_t`. I think this may be related to the cause of test failures on pre-5.0 servers.

Perhaps `MONGOC_SS_READ` and `has_write_stage` should be separate arguments to `_mongoc_cluster_stream_for_optype`?

Reply:
Good find. Implicit int-to-enum conversion strikes again.